### PR TITLE
Generate sugared client APIs

### PR DESCRIPTION
**Pull request by glbrntt**
Motivation:

The generated client API uses the full expressive request/response types
which are used by the interceptors. This gives clients full control but
at the cost of usability.

In many cases we can simplify this for them:

- For single requests we can push the message and metadata into the stub
  signature and construct the request on behalf of the user
- For streaming requests we can push the metadata and request writing
  closure into the signature of the stub and construct the request for
  the user
- For single responses we can default the response handler to returning
  the message

Modifications:

- Add sugared API
- This, required that the struct swift representation and renderer be
  updated to support dictionary literls
- Regenrate the code
- Update the echo example to use the convenience API

Result:

Easier to use client APIs

### CODE REVIEW

**Comment by glbrntt** [comment]
This is split into 3 commits:
1. [5602837](https://github.com/grpc/grpc-swift/pull/2009/commits/5602837590dccbb2190505662633c6209a7db0dd) the actual changes to the code generator and tests
2. [41c8f8c](https://github.com/grpc/grpc-swift/pull/2009/commits/41c8f8c459a5233509606170ae4e6c08ad82c37c) regenerating stubs
3. [ce47263](https://github.com/grpc/grpc-swift/pull/2009/commits/ce4726313243cc8cdfc132252d647a31dd34510c) updates the echo example to use the new APIs

File: Sources/GRPCCodeGen/Internal/Translator/ClientCodeTranslator.swift [hunk 1]
```diff
@@ -158,6 +151,225 @@ extension ClientCodeTranslator {
...
+      return .commentable(.preFormatted(method.documentation), .function(functionDescription))
+    }
+  }
+
+  private func makeParametersForSugaredMethodDeclaration(
+    method: CodeGenerationRequest.ServiceDescriptor.MethodDescriptor
+  ) -> [ParameterDescription] {
+    var parameters = [ParameterDescription]()
+
+    // Unary inputs have a 'message' parameters
```
Comment (gjcairo) [inline Sources/GRPCCodeGen/Internal/Translator/ClientCodeTranslator.swift:208]:
```suggestion
    // Unary inputs have a 'message' parameter
```

File: Sources/InteroperabilityTests/Generated/empty_service.grpc.swift [hunk 2]
```diff
@@ -77,6 +77,10 @@ public protocol Grpc_Testing_EmptyServiceClientProtocol: Sendable {}
 extension Grpc_Testing_EmptyService.ClientProtocol {
 }
 
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+extension Grpc_Testing_EmptyService.ClientProtocol {
+}
```
[the remark this answers is not in the document]
Comment (glbrntt) [reply Sources/InteroperabilityTests/Generated/empty_service.grpc.swift:82]: I don't think this should be done by the generator: if it's told to generate an empty extension it should. If anything it should be done by the translator but per another comment I don't think it's worth doing.

File: Sources/GRPCCodeGen/Internal/Renderer/TextBasedRenderer.swift [hunk 3]
```diff
@@ -528,6 +528,29 @@ struct TextBasedRenderer: RendererProtocol {
         writer.nextLineAppendsToLastLine()
       }
       writer.writeLine("]")
+
+    case .dictionary(let items):
```
[the remark this answers is not in the document]
Comment (gjcairo) [reply Sources/GRPCCodeGen/Internal/Renderer/TextBasedRenderer.swift:532]: I agree that we shouldn't be doing this at the text renderer level. Wondering if it's worth having a validation layer though, which I don't believe currently exists? (obviously out of scope for this)

**Review by gjcairo — APPROVED** [review]

File: Sources/GRPCCodeGen/Internal/Renderer/TextBasedRenderer.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (glbrntt) [reply Sources/GRPCCodeGen/Internal/Renderer/TextBasedRenderer.swift:532]: I don't think it'd be worth it. There is a _lot_ of validation we could do and we would almost certainly not have complete coverage. As an example it's very easy to build a structured Swift representation which is invalid. Technically we _could_ validate it and catch it ahead of time but the amount of code and effort to do that wouldn't be worth it. Our best bet at validation is simply using the generator on a variety of services.

**Pull request merged**